src/server/meme.js

```diff
@@ -18,4 +18,20 @@ module.exports = psql => ({
       return 'unexpected error when trying to save meme data';
     }
   },
+
+  getMemes: async (username) => {
+    try {
+      // get all memes if username not provided
+      if (username === '' || username === undefined || username === null) {
+        return await psql('memes');
+      }
+      // else get memes belonging to the provided user
+      return await psql('memes')
+        .innerJoin('users', 'memes.user_id', 'users.user_id')
+        .where('users.username', username);
```

Comment:
@fnoori I'd like to add a limit clause here like you did in your getComments PR. However in your PR you assume a limit field is sent in the request. I'm not sure I understand that, isn't it better to have a limit constant defined in meme.js?

Reply:
The reason why I added that on their is in case we ever wanted to change how many comments we would want to retrieve, since that's not the case, I updated my feature branch so it's a constant 20.

Also, I had to add a *temporary* retrieve memes handler in my branch. I did this because I needed it in order to get the live comments to work. Please see my comments branch, basically I needed to create a group for whoever is viewing the particular meme.

Reply:
@nxtman123 any suggestions on what the row limit for these queries should be?

Reply:
I think 15 at a time would work well. That's about 2.5 screens down at the least.

Reply:
@nxtman123 Wait, how will the client know which memes it hasn't received if the limit is set here? Should the client receive all memes, and display more as the user scrolls down?

Reply:
@adnanrh Checkout my comments branch and see how I retrieve the comments, knex provides an offset function which we can use to offset what to load next.

```
comments = await psql('comments')
    .where({ meme_id: data.meme_id })
    .orderBy('comment_id', 'desc')
    .limit(COMMENT_LIMIT)
    .offset(data.offset);
```

Reply:
@fnoori  Oh I see, thanks! Is a different offset persisted for each user though? Because there's only one psql connection

Reply:
The front end would need to handle incrementing the offset.
On the initial request, ```offset = 0```
After each subsequent request, the front end needs to ```offset += 20``` or whichever value we choose to increment by.

Reply:
I think we should do the same thing here as for comments. The client sends its earliest ID and the SQL query compares id < earliestId. By tracking offsets, especially here where latest memes aren't pushed to clients, it's likely that the offsets could get out of sync. Operating strictly by ID is more reliable.

Reply:
@nxtman123  @fnoori updated